R/get_undata.R

```diff
@@ -0,0 +1,177 @@
+
+  checkmate::assert_number(page)
+  checkmate::assert_list(query)
+
+  query[["v"]] <- NULL
+  query[["v"]] <- page
+
+  table <- polite::scrape(bow, query) %>%
+    rvest::html_element(css = ".DataContainer") %>%
+    rvest::html_table()
+
+  # TODO: create page number column
+
+  # TODO: merge on footnotes
+
+  checkmate::assert_tibble(table)
+  return(table)
+}
+
+#' @title Get the total number of pages available in the specified table.
+#'
+#' @inheritParams get_undata_table
+#'
+#' @return \[`numeric(1)`\] the total number of pages available in a table as
+#' specified by `query`.
+#'
+#' @family get_undata
+#'
+#' @importFrom magrittr %>%
+#' @export
+get_undata_npages <- function(bow, query) {
+  npages <- polite::scrape(bow, query) %>%
+    rvest::html_element("#spanPageCountB") %>%
+    rvest::html_text() %>%
+    as.numeric()
+  checkmate::assert_numeric(npages, len = 1)
+  return(npages)
+}
+
+#' @title Get the listed update dates for a data.un.org table
+#'
+#' @inheritParams get_undata_table
+#'
+#' @return a named \[`Date()`\] vector, the names correspond to the description
+#' of the date (likely 'Last' or 'Next').
+#'
+#' @family get_undata
+#'
+#' @importFrom magrittr %>%
+#' @export
+get_undata_update_dates <- function(bow, query) {
+
+  # get update element and separate lines
+  update_date_text <- polite::scrape(bow, query) %>%
+    rvest::html_element(".Update") %>%
+    rvest::html_text2() %>%
+    strsplit(split = "\n") %>%
+    unlist()
+
+  # get description of dates
+  update_dates_names <- update_date_text %>%
+    gsub(pattern = " update in UNdata: \\d*/\\d*/\\d*$", replacement = "")
+  checkmate::assert_character(update_dates_names, any.missing = FALSE)
+
+  # isolate actual dates
+  update_dates <- update_date_text %>%
+    gsub(
+      pattern = paste0("^(", paste(update_dates_names, collapse = "|"), ") update in UNdata: "),
+      replacement = ""
+    ) %>%
+    as.Date(format = "%Y/%m/%d")
+  checkmate::assert_date(update_dates, any.missing = FALSE)
+
+  names(update_dates) <- update_dates_names
+  return(update_dates)
+}
+
+#' @title Convert the undata url to the query argument needed to reference
+#'   specific tables
+#'
+#' @param table_url \[`character(1)`\]\cr
+#'   The full url to a specific table. If filters, columns, sort order or other
+#'   options are selected, use the 'Link to this page' -> 'Update' button to update
+#'   the `table_url`.
+#' @param base_url \[`character(1)`\]\cr
+#'   The base url of the data.un.org site. Default is 'http://data.un.org/Data.aspx'.
+#'
+#' @return \[`list()`\] of query parameters to pass to `get_undata_table` or
+#'   other related undata functions.
+#'
+#' @family get_undata
+#'
+#' @examples
+#' table_url <- "http://data.un.org/Data.aspx?d=PopDiv&f=variableID%3a12"
+#' query <- undata_url_to_query(table_url)
+#'
+#' @export
+undata_url_to_query <- function(table_url, base_url = "http://data.un.org/Data.aspx") {
```

Comment:
I appreciate that these are mostly very small functions. Makes it easier to follow. A vignette might be helpful though.